gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/oneconvergence_service_node_driver.py

```diff
@@ -816,7 +816,22 @@ def _update_template_with_firewall_rules(self, context, provider_ptg,
                     i += 1
 
         if consumer_cidr != '0.0.0.0/0' or not fw_rule_key:
-            stack_template[resources_key]['Firewall_Policy'][properties_key][
+            resource_name = 'OS::Neutron::FirewallPolicy'
+            fw_policy_key = self._get_heat_resource_key(
+                            stack_template[resources_key],
+                            is_template_aws_version,
+                            resource_name)
+            fw_resource_name = 'OS::Neutron::Firewall'
+            fw_key = self._get_heat_resource_key(
+                            stack_template[resources_key],
+                            is_template_aws_version,
+                            fw_resource_name)
+            # Include provider name in firewall, firewall policy.
+            ptg_name = '-' + provider_ptg['name']
+            stack_template[resources_key][fw_policy_key][properties_key]['name'] += ptg_name
+            stack_template[resources_key][fw_key][properties_key]['name'] += ptg_name
+
+            stack_template[resources_key][fw_policy_key][properties_key][
                 'firewall_rules'] = fw_rule_list
```

Comment:
replicated code from heat node driver. Re-use it from heat node driver by making it a common method